### PR TITLE
ci: doc-build: only rebase on pull requests, fix PDF generation again

### PR DESCRIPTION
**Pull request by fabiobaltieri**
> With that in mind, I think it's fair to keep it for now and re-evaluate next time the generation fails.

https://github.com/zephyrproject-rtos/zephyr/issues/64120#issuecomment-1795876979 foreshadowing...

---

Skip the rebase on non pull request run (there's no base), apply the container fix to the workflow, this is currently failing the PDF generation. Again.

### CODE REVIEW

**Comment by fabiobaltieri** [comment]
using the pr for some test while I'm at it

File: .github/workflows/doc-build.yml [hunk 1]
```diff
@@ -43,6 +43,10 @@ jobs:
       cancel-in-progress: true
 
     steps:
+    - name: Apply container owner mismatch workaround
+      run: |
+        git config --global --add safe.directory ${GITHUB_WORKSPACE}
```
Comment (fabiobaltieri) [inline .github/workflows/doc-build.yml:48]: this is not really needed since this specific step does not use containers but I guess it does not hurt either

Comment (fabiobaltieri) [reply .github/workflows/doc-build.yml:48]: no screw that, I'll drop it

**Review by kartben — APPROVED** [review]
Thank you for your service! 🫡

**Comment by fabiobaltieri** [comment]
> Thank you for your service! 🫡

Anything to ensure the continuous silky smooth operation of our extremely valuable PDF generation service. 😶 How about we stop generating the HTML documentation?

**Review by gmarull — APPROVED** [review]

**Comment by gmarull** [comment]
we'll soon need a supercomputer to build PDF, start preparing the fix.

**Comment by fabiobaltieri** [comment]
Hey, merging this so it hits the next scheduled run, one of the PR run had the PDF job enabled and it worked so we should be good to go.

**Pull request merged**